### DISCUSSION TITLE
[Solved] sonoff snzb-02D not reporting battery

### DISCUSSION THREAD

**Discussion by christescu** [Koenkk/zigbee2mqtt#16390] [Devices]
Sonoff SNZB-02D just get supported out of the box.
Temperature and humidity are reported fine, however battery is not reported.

**Comment by christescu** [comment]
Later edit:
After one day, battery is reported as well.

**Comment by GryKyo** [comment]
I have paired this sensor (SNZB-02D) many times to my zigbee2mqtt running in a docker container and it always shows up as unsupported?  As far as I know I have pulled and am running the latest image.  Is there somewhere I can add the sensor manually?

FYI:  The sensor will pair first try but shows in the front end as "Unsupported".  I can see a MQTT message and a popup on the front end every time the sensor reports values but there is no value exposed?

Thanks
Garry

**Reply by christescu** [reply]
You have to update  zigbee2mqtt to last version, in my case it is 1.29.2.
Or, you can copy sonoff.js from:
https://github.com/Koenkk/zigbee-herdsman-converters/tree/master/devices
and create an external converter as described here:
https://www.zigbee2mqtt.io/advanced/support-new-devices/01_support_new_devices.html

**Comment by pressla** [comment]
It does identify the device correctly, and lists all values, but I get a configuration error 2023-02-01 14:59:16Configure failed: Error: ConfigureReporting 0x30fb10fffe6e99bc/1 genPowerCfg([{"attribute":"batteryVoltage","minimumReportInterval":3600,"maximumReportInterval":62000,"reportableChange":0}], {"sendWhen":"active","timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":true,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":null,"transactionSequenceNumber":null,"writeUndiv":false}) failed (Status 'UNSUPPORTED_ATTRIBUTE')
and none of the values gets updated then.
